api/prometheus/v1/api.go

```diff
@@ -845,6 +883,10 @@ func (h *httpAPI) Query(ctx context.Context, query string, ts time.Time, opts ..
 		q.Set("timeout", d.String())
 	}
 
+	if opt.perStepStats {
```

Comment:
I think it would be nicer to stick to some enum like `all` and `minimal` instead of `opt.perStepStats` bool - we will be resilient to adding any other option.

Reply:
It's mostly boolean because I based this on the PR for Prometheus that adds it, and it had a bool - https://github.com/prometheus/prometheus/blob/ce6a643ee88fba7c02fbd0459c4d0ac498f512dd/promql/engine.go#L130

The only supported value here is "all", at least at the moment.
https://github.com/prometheus/prometheus/blob/a64b9fe323cf913c212c1829fe93d1a9dbc63348/web/api/v1/api.go#L433
Not sure if it's worth to create a client abstraction for it but happy to do so.

IMHO ideally it should be an enum in Prometheus source, since that's where it's consumed and used, but I imagine we want to avoid prometheus & client_golang circular dependencies, so it seems like we'll need to duplicate supported values in client_golang.

Side note: I do find it awkward at time to use client_golang to interact with Prometheus, it's easy for client_golang to fall behind Prometheus (structs for JSON might not get updated, #925) so I wonder if in the future it would make sense to move API client libs to Prometheus repo itself, especially now that it's versioned better and so it's easier to import and use.